ubi_manifest/worker/tasks/depsolver/rpm_depsolver.py

```diff
@@ -207,19 +207,34 @@ def resolve_rpms(
         return self.what_provides(batch, "provides.name", repos, blacklist)
 
     def get_source_pkgs(
-        self, binary_rpms: list[UbiUnit], blacklist: list[PackageToExclude]
+        self,
+        binary_rpms: set[UbiUnit],
+        binary_repos: list[YumRepository],
+        blacklist: list[PackageToExclude],
     ) -> set[UbiUnit]:
-        crit = create_or_criteria(
-            ["filename"], [(rpm.sourcerpm,) for rpm in binary_rpms if rpm.sourcerpm]
-        )
-
+        srpm_repos = set()
+        filenames = set()
+
+        for repo in binary_repos:
+            if not repo.relative_url:
+                continue
+            for rpm in binary_rpms:
+                if rpm.sourcerpm and rpm.associate_source_repo_id == repo.id:
+                    srpm_repo = repo.get_source_repository().result()
+                    if srpm_repo:
+                        srpm_repos.add(srpm_repo)
+                        filenames.add(rpm.sourcerpm)
```

Comment:
OK, so this will fulfill the one part of AC:
`ubi-manifest searches for required SRPMs only once and doesn't do any search for already found SRPMs`
but I don't see where the next AC is implemented:
`SRPM is search only in directly related source RPM repo to its binary variant where non-srpm package live.`